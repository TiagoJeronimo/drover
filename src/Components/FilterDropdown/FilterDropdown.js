import React from 'react';
import { Dropdown } from 'semantic-ui-react';
import './FilterDropdown.style.scss';

const FilterDropdown = ({
  filterName,
  labelText,
  dropdownOptions,
  handleFilterChange,
  selectedFilter
}) => (
  <div className="filterDropdown-container">
    <div className="filterDropdown-label">{labelText}</div>
    <Dropdown
      placeholder="Any"
      fluid
      floating
      selection
      text={selectedFilter}
      name={filterName}
      options={dropdownOptions}
      onChange={handleFilterChange}
    />
  </div>
);

export default FilterDropdown;
