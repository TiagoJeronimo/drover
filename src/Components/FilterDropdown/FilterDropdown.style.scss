
.filterContainer {
    width: 100%;
    min-height: 25px;
    margin-bottom: 16px;
}

.filterLabel {
    margin: 10px 0;
    font-size: 16px;
}