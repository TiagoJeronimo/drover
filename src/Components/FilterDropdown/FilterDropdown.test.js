import React from 'react';
import { mount } from 'enzyme';

import FilterDropdown from './FilterDropdown';

it('FilterDropdown with normal props', () => {
  const props = {
    filterName: 'vehicle_make',
    labelText: 'Car Make',
    dropdownOptions: [{ text: 'Any', value: 'any' }, { text: 'Car', value: 'car' }],
    handleFilterChange: jest.fn(),
    selectedFilter: 'Any'
  };
  const component = <FilterDropdown {...props} />;
  const html = mount(component).html();
  expect(html).toMatchSnapshot();
});
