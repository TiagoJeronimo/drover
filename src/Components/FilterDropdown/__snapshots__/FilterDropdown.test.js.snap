// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDropdown with normal props 1`] = `"<div class=\\"filterDropdown-container\\"><div class=\\"filterDropdown-label\\">Car Make</div><div name=\\"vehicle_make\\" role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui fluid floating selection dropdown\\" tabindex=\\"0\\"><div class=\\"default text\\" role=\\"alert\\" aria-live=\\"polite\\">Any</div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"true\\" class=\\"selected item\\"><span class=\\"text\\">Any</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Car</span></div></div></div></div>"`;
